asllib/doc/ASLTypingReference.tex

```diff
@@ -425,480 +813,582 @@ \chapter{Reading guide}
-    \item integer;
-    \item real;
-    \item string;
-    \item boolean;
-    \item bits;
-    \item enumeration.
-    \end{itemize}
+\subsection{Prose}
+The \emph{builtin singular types} are:
+\begin{itemize}
+\item \texttt{integer};
+\item \texttt{real};
+\item \texttt{string};
+\item \texttt{boolean};
+\item \texttt{bits} (which also represents \texttt{bit}, as a special case);
+\item \texttt{enumeration}.
+\end{itemize}
 
-    \subsection{Example: TypingRule.BuiltinSingularTypes.asl}
+\subsection{Example: TypingRule.BuiltinSingularTypes.asl}
 
 In this example:
 \VerbatimInput[firstline=3,lastline=8]{\tests/TypingRule.BuiltinSingularTypes.asl}
 
 Variables of builtin singular types \texttt{integer}, \texttt{real},
 \texttt{boolean}, \texttt{bits(4)}, \\ and~\texttt{bits(2)} are defined.
 
-    \subsection{Example: TypingRule.EnumerationType.asl}
+\subsection{Example: TypingRule.EnumerationType.asl}
 \VerbatimInput{\tests/TypingRule.EnumerationType.asl}
 The builtin singular type \texttt{color} consists in two constants
 \texttt{RED}, and~\texttt{BLACK}.
 
-    \subsection{Code}
-      \VerbatimInput[firstline=\BuiltinSingularBegin, lastline=\BuiltinSingularEnd]{../types.ml}
+\CodeSubsection{\BuiltinSingularBegin}{\BuiltinSingularEnd}{../types.ml}
 
-\begin{formal}
-   \subsection{Formally}
+\begin{emptyformal}
+\subsection{Formally}
 \begin{mathpar}
 \inferrule{
-  \astlabel(\tty) \in \{\TReal, \TString, \TBool, \TBits, \TEnum, \TInt\}
+  \vb \eqdef \astlabel(\tty) \in \{\TReal, \TString, \TBool, \TBits, \TEnum, \TInt\}
 }{
-  \isbuiltinsingular(\tty)
+  \isbuiltinsingular(\tty) \typearrow \vb
 }
 \end{mathpar}
+\end{emptyformal}
 
-\end{formal}
-
-    \subsection{Comments}
-    This is related to \identd{PQCK} and \identd{NZWT}.
+\subsection{Comments}
+This is related to \identd{PQCK} and \identd{NZWT}.
 
 \section{TypingRule.BuiltinAggregateType \label{sec:TypingRule.BuiltinAggregateType}}
+\hypertarget{def-isbuiltinaggregate}{}
+The predicate
+\[
+  \isbuiltinaggregate(\overname{\ty}{\tty}) \;\aslto\; \Bool
+\]
+tests whether the type $\tty$ is a \emph{builtin aggregate type}.
 
-    \subsection{Prose}
-    The builtin aggregate types are:
-    \begin{itemize}
-    \item tuple;
-    \item array;
-    \item record;
-    \item exception.
-    \end{itemize}
+\subsection{Prose}
+The builtin aggregate types are:
+\begin{itemize}
+\item tuple;
+\item \texttt{array};
+\item \texttt{record};
+\item \texttt{exception}.
+\end{itemize}
 
-    \subsection{Example: TypingRule.BuiltinAggregateTypes.asl}
-      \VerbatimInput{\tests/TypingRule.BuiltinAggregateTypes.asl}
-      Type \texttt{pair} is the type of integer and booleans pairs. Notice that
-      the syntax of types and expressions are similar.
+\subsection{Example: TypingRule.BuiltinAggregateTypes.asl}
+\VerbatimInput{\tests/TypingRule.BuiltinAggregateTypes.asl}
+Type \texttt{pair} is the type of integer and booleans pairs. Notice that
+the syntax of types and expressions are similar.
```

Comment:
similar, in what sense? please expand and give example

Reply:
I don't know who wrote this. Anyway, it doesn't make sense to me so I'm removing this comment.